backend/pkg/dbstore/report_group_membership.go

```diff
@@ -44,51 +55,89 @@ func (datasource *SQLiteDatasource) GroupMemberUserIDs(reportGroup ReportGroup)
 	return userIDs, nil
 }
 
-func (datasource *SQLiteDatasource) GetReportGroupMemberships(groupID string) []ReportGroupMembership {
-	db, _ := sql.Open("sqlite3", datasource.Path)
+func (datasource *SQLiteDatasource) GetReportGroupMemberships(groupID string) ([]ReportGroupMembership, error) {
+	db, err := sql.Open("sqlite3", datasource.Path)
 	defer db.Close()
+	if err != nil {
+		log.DefaultLogger.Error("GetReportGroupMemberships: sql.Open(): ", err.Error())
+		return nil, err
+	}
 
 	var memberships []ReportGroupMembership
 
-	rows, _ := db.Query("SELECT * FROM ReportGroupMembership WHERE reportGroupID = ?", groupID)
+	rows, err := db.Query("SELECT * FROM ReportGroupMembership WHERE reportGroupID = ?", groupID)
 	defer rows.Close()
+	if err != nil {
+		log.DefaultLogger.Error("GetReportGroupMemberships: db.Query(): ", err.Error())
+		return nil, err
+	}
 
 	for rows.Next() {
 		var ID, UserID, ReportGroupID string
-		rows.Scan(&ID, &UserID, &ReportGroupID)
+		err = rows.Scan(&ID, &UserID, &ReportGroupID)
+		if err != nil {
+			log.DefaultLogger.Error("GetReportGroupMemberships: rows.Scan(): ", err.Error())
+			return nil, err
+		}
 		membership := ReportGroupMembership{ID, UserID, ReportGroupID}
 		memberships = append(memberships, membership)
 	}
 
-	return memberships
+	return memberships, nil
 }
 
 func (datasource *SQLiteDatasource) CreateReportGroupMembership(members []ReportGroupMembership) ([]ReportGroupMembership, error) {
-	db, _ := sql.Open("sqlite3", datasource.Path)
+	db, err := sql.Open("sqlite3", datasource.Path)
 	defer db.Close()
+	if err != nil {
+		log.DefaultLogger.Error("CreateReportGroupMembership: sql.Open(): ", err.Error())
+		return nil, err
+	}
 
 	var addedMemberships []ReportGroupMembership
```

Comment:
A little off-road, but I guess good as any time. Wonder if since you know the `range members` up front, if you could use an array instead of a slice here? Probably wouldn't gain anything... but I read about arrays vs. slices so wanted to use what I learned somewhere!

Reply:
yeah def - just what I was using, open to seeing different ways (in code)

Reply:
I *think* it's pretty much the same, just instead of:

`var addedMemberships []ReportGroupMembership`

something like:

```
var membersCount = range members
var addedMemberships [membersCount]ReportGroupMembership
```

Untested, but I think it's basically just fixed size versus dynamic (so you prob get a bit of speed from reserving fixed memory space, but at the cost of flexibility)?